oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/alarm/AlarmRecord.java

```diff
@@ -78,6 +78,7 @@ public StorageID id() {
     private String id0;
     @Column(name = ID1, storageOnly = true)
     private String id1;
+    @ElasticSearch.EnableDocValues
```

Comment:
Why does alarm start time need this?

Reply:
> Why does alarm start time need this?

This reminds me one potential issue here, if we restrict to only add fields used in this repo, it might break third parties’s custom plugin if they add more features to their own plugin by aggregating/sorting some fields that we didn’t enable doc_values

Reply:
Yes, it will. Generally, it may be worth to see how much benefit we will get from this change. Could you try a benchmark about this?

Reply:
But to be honest, we never guarantee users could read data from elasticsearch on their own, we only guarantee from our GraphQL/PromQL.

Reply:
> Yes, it will. Generally, it may be worth to see how much benefit we will get from this change. Could you try a benchmark about this?

This doesn’t really have much benefit for us in my opinion, all the goods disabling `doc_values` brings is disk space, from the doc, I’m not sure whether minimizing disk space is our goal.

> Disabling doc values
> 
> All fields which support doc values have them enabled by default. If you are sure that you don’t need to sort or aggregate on a field, or access the field value from a script, you can disable doc values in order to save disk space:

Reply:
> But to be honest, we never guarantee users could read data from elasticsearch on their own, we only guarantee from our GraphQL/PromQL.

OK that’s fair enough, I think we can safely do so if that’s not guaranteed

Reply:
This proposal was from @hanahmily about reducing the cost of trace/log queries. I believe. 

That was why I asked for a disabling annotation only. That makes things easier.

Reply:
> This proposal was from @hanahmily about reducing the cost of trace/log queries. I believe.

I think that’s not what disabling the `doc_values` can achieve, https://www.elastic.co/guide/en/elasticsearch/reference/current/doc-values.html doesn’t mentioned disabling `doc_values` can speed up query performance at all.

Reply:
@hanahmily can you share where you find disabling `doc_values` can speed up the query performance?

Reply:
No, he was talking about BanyanDB, and the concept was from this config.

Reply:
The key use case is when we talk about `_traffic` indices and trace/log indices. Conditions for that, is not used for sorting and aggregation. So, we could reduce the payload. 

```
1. Value column of metrics.
2. Conditions of logs and traces(skywalking and zipkin) exclude latency and timestamp, which are used in sorting.
3. All searchable field in metadata(*_traffic)
```

In the original issue, I only proposed three use cases. Nothing more in my mind.

Reply:
> Conditions for that, is not used for sorting and aggregation. So, we could reduce the payload.

What I’m wondering is this, what “payload” are we trying to reduce, as mentioned, disabling `doc_values` is mainly for reducing disk space, I don’t see how it would speed up in terms of query performance like you said here https://github.com/apache/skywalking/pull/12782#discussion_r1851525822. If reducing disk space is our goal, disabling all possible fields will maximize the outcome, that’s why I tend to disable doc_values by default and opt in those fields that need this feature.